lib/googleauth/credentials.rb

```diff
@@ -355,7 +355,7 @@ def initialize keyfile, options = {}
         @project_id = options["project_id"] || options["project"]
         @quota_project_id = options["quota_project_id"]
         case keyfile
-        when Signet::OAuth2::Client
+        when Signet::OAuth2::Client, Google::Auth::BaseClient
```

Comment:
Should be able to remove `Signet::OAuth2::Client` from this matcher, since `Signet::OAuth2::Client` now includes `Google::Auth::BaseClient`.